### PR TITLE
Differentiate between a failing/finished PBS job

### LINKED ISSUE #10680: Fail timeout jobs instead of labeling them as finished
## Rationale
The are_queued_jobs_finished.py helper script needs to support a failure when PBS decides a job has failed. In this case, when walltime is exceeded.

## Description
When using are_queued_jobs_finished.py,  and the script encounters a finished job due to the job exceeding walltime, the script sets a finished status... as normal. The problem is, the TestHarness is instructed to _not_ ask for additional details about finished jobs (such as making a call to qstat). So the TestHarness has no idea that the job in question actually failed, and thus, sends the job along to the tester to process results. And in the case of many tests, where only certain output discerns a test as a failure, that output may not be found (because PBS killed the job early), and thus a test which should be failing, is now passing.

## Impact
None.

### PR DESCRIPTION
**Pull request by milljm**
We can't set a finished 'waiting' status for jobs
killed by PBS. We must set a failed status accordingly
for jobs killed by PBS.

Closes #10680

### CODE REVIEW

**Comment by milljm** [comment]
Tag @gardnerru

File: scripts/are_queued_jobs_finished.py [hunk 1]
```diff
@@ -45,21 +46,34 @@ def getQueuedJobIDs(jobs):
...
+    return results
 
 def updateStatus(jobs, finished_jobs):
     """ loop through jobs and adjust the status for those supplied in finished_jobs """
     for job, meta in jobs.getJobs().iteritems():
-        if 'job_id' in meta and meta['job_id'] in finished_jobs and meta['status_bucket']['status'] == 'QUEUED':
-            meta['status_bucket']['status'] = 'WAITING'
+        # Use list splicing to simplify the search process
+        if 'job_id' in meta \
+           and meta['job_id'] in finished_jobs[0::2] \
```
Comment (brianmoose) [inline scripts/are_queued_jobs_finished.py]: Why not make `finished_jobs` a dict with `job_id` as the key and the status as the value? Then you wouldn't have to do the stride splicing.

**Review by permcody — COMMENTED** [review]
Looks OK, but this is all a big bandaid over the real problem. When are you going to try implementing "statuses" and "outcomes"?

File: scripts/are_queued_jobs_finished.py [hunk 2]
```diff
@@ -45,21 +46,34 @@ def getQueuedJobIDs(jobs):
...
+    t.close()
+    return results
 
 def updateStatus(jobs, finished_jobs):
     """ loop through jobs and adjust the status for those supplied in finished_jobs """
     for job, meta in jobs.getJobs().iteritems():
-        if 'job_id' in meta and meta['job_id'] in finished_jobs and meta['status_bucket']['status'] == 'QUEUED':
-            meta['status_bucket']['status'] = 'WAITING'
+        # Use list splicing to simplify the search process
+        if 'job_id' in meta \
```
Comment (brianmoose) [inline scripts/are_queued_jobs_finished.py]: I think the recommended way to have a multiline if is to use `if ()`

File: scripts/are_queued_jobs_finished.py [hunk 3]
```diff
@@ -68,11 +82,21 @@ def isFinished(jobs):
...
+                job_id = re.search(r'^Job Id: (\d+)\.[\W\w]+$', job).group(1)
+                job_state = re.search(r'job_state = (\w)', job).group(1)
+                if job_state == 'F':
+                    job_status = {'status' :'WAITING', 'color' : 'CYAN' }
+
+                    # if exit status is non-zero, fail the job
+                    if int(re.search(r'Exit_status = (\d+)', job).group(1)):
+                        job_status = { 'status' :'FAIL', 'color' : 'RED' }
+
+                    finished_jobs.extend([job_id, job_status])
```
Comment (brianmoose) [inline scripts/are_queued_jobs_finished.py]: Might be nicer to just have `finished_jobs[job_id] = job_status` so you don't have to do the stride splicing above.

**Review by brianmoose — CHANGES_REQUESTED** [review]

File: scripts/are_queued_jobs_finished.py (hunk 1 above)
Comment (milljm) [reply scripts/are_queued_jobs_finished.py]: I thought about doing it that way in the beginning... I'll take another stab at it. But I hit a road block somewheres...

**Comment by milljm** [comment]
@permcody it certainly is a bandaid. But the status overhaul PR is a long ways from being usable at the moment. It will be a _huge_ scheduler change, which historically will not merge any time soon.

**Comment by milljm** [comment]
Hold off on this, until we get the new status system in place.

**Pull request closed without merging**